src/components/BusinessDonationHistory/BusinessDonationHistory.jsx

```diff
@@ -11,17 +25,45 @@ const BusinessDonationHistory = () => {
   const [selectedDonationId, setSelectedDonationId] = useState(null);
   const TABLE_HEADERS = ['Submitted By', 'Food Bank', 'Date', 'Action'];
 
+  // for pagination
+  const [currentTotalDonationNum, setCurrentTotalDonationNum] = useState(0);
+  const [currentPageNum, setCurrentPageNum] = useState(1);
+  const [pageLimit, setPageLimit] = useState(1);
+
+  // for search
+  const [searchInput, setSearchInput] = useState('');
+
+  // TEMPORARY DISABLE LINE SO THAT IN THE FUTURE IF YOU WANT TO ADD
+  // A DROP DOWN TO CHANGE PAGINATION NUMBER JUST USE THE SET FUNCTION
+  // AND REMOVE THE COMMENT
+  // eslint-disable-next-line no-unused-vars
+  const [paginationNumber, setPaginationNumber] = useState(13);
+
   useEffect(() => {
     const getData = async () => {
       try {
-        const response = await backend.get('/donation');
+        const response =
+          searchInput === ''
+            ? await backend.get(
+                `/donation/?donationsLimit=${paginationNumber}&pageNum=${currentPageNum}`,
+              )
+            : await backend.get(
+                `/donation/search/${searchInput}?donationsLimit=${paginationNumber}&pageNum=${currentPageNum}`,
+              );
+
         setData(response.data);
+
+        const donationNumResponse = await backend.get(
+          `/donation/totalDonations/${searchInput === '' ? '' : searchInput}`,
+        );
+        setCurrentTotalDonationNum(donationNumResponse.data[0]['count']);
+        setPageLimit(Math.ceil(donationNumResponse.data[0]['count'] / paginationNumber));
       } catch (error) {
         console.error('Error fetching data:', error);
       }
     };
     getData();
-  }, []);
+  }, [backend, currentPageNum, searchInput]);
```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useEffect has a missing dependency: 'paginationNumber'. Either include it or remove the dependency array. You can also replace multiple useState variables with useReducer if 'setPageLimit' needs the current value of 'paginationNumber'.

Reply:
add paginationNumber to dependencies